.evergreen/build-mac.sh

```diff
@@ -12,6 +12,14 @@ for VERSION in 3.7 3.8 3.9 3.10; do
     PYTHON=/Library/Frameworks/Python.framework/Versions/$VERSION/bin/python3
     rm -rf build
 
+    # Set the arch flags appropriately for the target platform.
+    PLATFORM=$($PYTHON -c "import sysconfig;print(sysconfig.get_platform())")
+    if [[ $PLATFORM =~ "universal2" ]]; then
+        ARCHFLAGS="-arch arm64 -arch x86_64"
+    else
+        ARCHFLAGS="-arch x86_64"
+    fi
```

Comment:
Is there any point when we can remove this hack? It seems like something that should happen automatically.

Reply:
We don't actually need this hack, I just haven't pushed yet because I'm using direct patch builds.